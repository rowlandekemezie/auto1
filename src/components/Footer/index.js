import React from "react";
import styled from "styled-components";

const Wrapper = styled("div")`
  flex: 0 0 80px;
  border-top: 2px solid ${({ theme }) => theme.GRAY};
  text-align: center;
`;

function Footer() {
  return (
    <Wrapper>
      <p>&copy; AUTO1 Group 2019</p>
    </Wrapper>
  );
}

export default Footer;
