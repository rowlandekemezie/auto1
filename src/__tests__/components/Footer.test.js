import React from "react";
import { shallow, mount } from "enzyme";
import Footer from "../../components/Footer";

describe("<Footer />", () => {
  it("should render component properly", () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correct text in the footer", () => {
    const wrapper = mount(<Footer />);
    expect(wrapper.find("div").length).toBe(1);
    expect(wrapper.find("p").length).toBe(1);
    expect(wrapper.find("p").text()).toMatch(new RegExp(/AUTO1 Group 2019/));
  });
});
