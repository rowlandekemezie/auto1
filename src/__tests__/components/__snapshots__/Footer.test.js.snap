// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render component properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <p>
        © AUTO1 Group 2019
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "© AUTO1 Group 2019",
      },
      "ref": null,
      "rendered": "© AUTO1 Group 2019",
      "type": "p",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  flex: 0 0 80px;
  border-top: 2px solid ",
          [Function],
          ";
  text-align: center;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <p>
          © AUTO1 Group 2019
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "© AUTO1 Group 2019",
        },
        "ref": null,
        "rendered": "© AUTO1 Group 2019",
        "type": "p",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  flex: 0 0 80px;
  border-top: 2px solid ",
            [Function],
            ";
  text-align: center;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
